### ISSUE TITLE
Split Tracker good/useless item categorization into three

### ISSUE THREAD

**Issue by Vivelin** [Vivelin/SMZ3Randomizer#122]
Currently, deciding whether an item is good ("progression") is largely arbitrary and depends exclusively on the item type itself.

Some effort was made to extend this to include more logic in 719153e88bed8c53c7feae48eecbda6c04f3cd3e, but it is ultimately still mostly flawed because it relies on whether the item at the end of the chain is "good" or not.

My suggestion would be to split this into three categories:
- **Strictly required** for go mode. This depends on logic and currently tracked items.
  Examples: Morph Ball; Pegasus Boots when GT Big Key is on Bob's Torch; the first (two) Power Bombs.
- **Nice to have** but technically skippable. For example, mail upgrades.
- Everything else, like missiles, arrows, rupees, etc.

**Comment by Vivelin** [comment]
@MattEqualsCoder Should we make this configurable? 

I imagine what's useful differs mostly based on skill level (and seed, of course. Maybe we could also factor that in without spoiling somehow?). Betus getting baited by hints and ending up with a sword is something different from, for example, me playing.

**Comment by MattEqualsCoder** [comment]
Yeah, this is pretty tricky because some things are sort of fluid and dependent on progression. For example, level 1 and 2 swords are required for checks and should be labeled as such. Levels 3 and 4 would be nice to have. Shields generally are not important, but they can be technically required if you don't get the cape or cane of byrna for TR.

I think for swords it may just be best to code something special for them sadly to factor that in. Other than that, I almost wonder if we need to have 4 categories:

- **Almost certainly required** - These are things that you tend to need for every single seed. Level 1 sword, morph ball, gloves, etc.
- **Possibly required** - This would be for checks that will open up a few locations, but could be skippable depending on the circumstances. Cape, level 2 sword, bottle, wave beam, etc. Keys might go in here in keysanity? But honestly there are so many keys I could almost see the argument of them being their own hint category.
- **Nice to have** - Same as above. I'd maybe add maps to this category if in keysanity.
- **Everything else** - Same as above.

Items could be specifically placed in these categories, but another option could be that we could do something similar to how progression item locations are picked. Basically we could take the player's current inventory, give them all other potential progression items *except* for the item(s) at the location(s) and see if there are any locations that would be opened up if you give the player that item. If it opens up like 1-5 locations or something, it'll be stated to be possibly required. If it's more, we say it's almost certainly required.

One thing that makes this solution nice is that it could factor in people doing things out of logic as well as account for multiple options for item requirements. For example, early on it may say that the speedbooster is almost certainly required, but later on if people mockball, get other items that make the speedbooster not as important, etc. it could be down to there just being a handful of locations that are left that could require it and it would be listed as possibly required.

**Comment by MattEqualsCoder** [comment]
So I think with the hint changes, I'm not sure if this is necessary anymore?
